src/modules/fw_pos_control/performance_model/PerformanceModel.cpp

```diff
@@ -0,0 +1,225 @@
+		climbrate_max = math::max(_param_fw_t_clmb_max.get() + density_gradient * delta_rho, kClimbrateMin);
+	}
+
+	return climbrate_max / getWeightRatio();
+}
+float PerformanceModel::getTrimThrottle(float throttle_min, float throttle_max, float airspeed_sp,
+					float air_density) const
+{
+	const float throttle_trim = getTrimThrottleForAirspeed(airspeed_sp) * getAirDensityThrottleScale(
+					    air_density) * getWeightThrottleScale();
+	return math::constrain(throttle_trim, throttle_min, throttle_max);
+}
+float PerformanceModel::getWeightThrottleScale() const
+{
+	return sqrtf(getWeightRatio());
+}
+
+float PerformanceModel::getAirDensityThrottleScale(float air_density) const
+{
+	float air_density_throttle_scale = 1.0f;
+
+	if (PX4_ISFINITE(air_density)) {
+		// scale throttle as a function of sqrt(rho0/rho)
+		const float eas2tas = sqrtf(CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C / air_density);
+		const float eas2tas_at_5000m_amsl = sqrtf(CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C / kAirDensityStandardAtmos5000Amsl);
+		air_density_throttle_scale = math::constrain(eas2tas, 1.f, eas2tas_at_5000m_amsl);
+	}
+
+	return air_density_throttle_scale;
+}
+float PerformanceModel::getTrimThrottleForAirspeed(float airspeed_sp) const
+{
+	float throttle_trim =
+		_param_fw_thr_trim.get(); // throttle required for level flight at trim airspeed, at sea level (standard atmosphere)
+
+	// Drag modelling (parasite drag): calculate mapping airspeed-->throttle, assuming a linear relation with different gradients
+// above and below trim. This is tunable thorugh FW_THR_ASPD_MIN and FW_THR_ASPD_MAX.
+	const float slope_below_trim = (_param_fw_thr_trim.get() - _param_fw_thr_aspd_min.get()) /
+				       (_param_fw_airspd_trim.get() - _param_fw_airspd_min.get());
+	const float slope_above_trim = (_param_fw_thr_aspd_max.get() - _param_fw_thr_trim.get()) /
+				       (_param_fw_airspd_max.get() - _param_fw_airspd_trim.get());
+
+	if (PX4_ISFINITE(airspeed_sp) && PX4_ISFINITE(slope_below_trim) && _param_fw_thr_aspd_min.get() > FLT_EPSILON
+	    && airspeed_sp < _param_fw_airspd_trim.get()) {
+		throttle_trim = _param_fw_thr_trim.get() - slope_below_trim * (_param_fw_airspd_trim.get() - airspeed_sp);
+
+	} else if (PX4_ISFINITE(airspeed_sp) && PX4_ISFINITE(slope_above_trim) && _param_fw_thr_aspd_max.get() > FLT_EPSILON
+		   && airspeed_sp > _param_fw_airspd_trim.get()) {
+		throttle_trim = _param_fw_thr_trim.get() + slope_above_trim * (airspeed_sp - _param_fw_airspd_trim.get());
+	}
+
+	return throttle_trim;
+}
+float PerformanceModel::getMinimumSinkRate(float air_density) const
+{
+	return _param_fw_t_sink_min.get() * sqrtf(getWeightRatio() * CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C / air_density);
+}
+float PerformanceModel::getTrimAirspeed() const
+{
+	return math::constrain(_param_fw_airspd_trim.get() * sqrtf(getWeightRatio()), _param_fw_airspd_min.get(),
+			       _param_fw_airspd_max.get());
+}
+float PerformanceModel::getMinimumAirspeed(float load_factor) const
```

Comment:
Should we specify in the function name that it's about CAS and not TAS? As otherwise it's not clear why it's not adjusted for air density as well.

Reply:
or include the specification in the function brief if you dont want to make the function name longer

Reply:
It was already in the brief, I added it now to all function names that return a calibrated airspeed.